File: 백준/Gold/1253. 좋다/좋다.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
)

func remove(s []int, index int) []int {
	ret := []int{}
	ret = append(ret, s[:index]...)
	return append(ret, s[index+1:]...)
}
func main() {
	reader := bufio.NewReader(os.Stdin)
	var n, good int
	fmt.Fscanf(reader, "%d\n", &n)
	var numbers = make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Fscanf(reader, "%d ", &numbers[i])
	}
	sort.Ints(numbers)
	var left, right, sum int
	for i := 0; i < n; i++ {
		targets := remove(numbers, i)
		left, right = 0, len(targets)-1
		for left < right {
			sum = targets[left] + targets[right]
			if sum == numbers[i] {
				good += 1
				break
			} else if sum < numbers[i] {
				left++
			} else {
				right--
			}
		}
	}
	fmt.Println(good)
}

File: 백준/Gold/1253. 좋다/좋다.py
import sys
from collections import deque
input=sys.stdin.readline
sys.setrecursionlimit(10**6)
n = int(input())
nums = list(map(int, input().split()))
good = 0
nums.sort()
for i in range(n):
    targets = nums[:i] + nums[i+1:]
    left, right = 0, len(targets)-1
    while left < right:
        sum = targets[left] + targets[right]
        if sum == nums[i]:
            good += 1
            break
        elif sum < nums[i]:
            left += 1
        else:
            right -= 1
print(good)